### ISSUE TITLE
PS move and eye don't work after build

### ISSUE THREAD

**Issue by hazem17** [HipsterSloth/psmove-unity5#19]
When i build the game on windows it doesn't work at all, showing the debug the error message 
"failed to load library assets/plugins/x86_64/psmoveapi.dll" shows, i tried to build 32bit but the same error message shows even if i copied the dlls to the plugin folder in the build, i also tried to copy the dlls to a folder named "x86_64" it then crashes the whole build.
can you help me

**Comment by Czeus** [comment]
I also have this problem. Could you help us?

**Comment by HipsterSloth** [comment]
So the version of psmoveapi I built is 64-bit (so that it will run in the editor. I can build a 32-bit version of the dll as well this weekend. Sorry I hadn't replied to this issue in a while (out of town on business).

**Comment by Ideae** [comment]
This is also an issue for me. I'll be awaiting an update on this, thanks! :)

**Comment by HipsterSloth** [comment]
Ok. I believe I finally have this fixed with this check-in 18dd8e13bc6bcbeeca14eaa47b1c2c33e1015446: 
* I made a 32-bit build of my fork of [psmoveapi](https://github.com/HipsterSloth/psmoveapi/tree/psmove_unity5) and put the newly built dlls/utilities in assets/plugins/x86
* Fixed some path issues in the dll loading code that was causing the manual dll loading to not work in standalone builds
* Tested in editor (which is 64-bit on my machine) and with x86 and x86_64 standalone builds.

Sorry this took me so long to getting around to fixing. Please let me know if this fix works or not for you guys. If all looks good I'll close this issue.

**Comment by Ideae** [comment]
Thanks for the commit. Unfortunately, it's not working for me. (Win10, Unity 5.3.41f)
When I run the built demo scene in this repository, it crashes. It still works running in editor mode. I'm going to email you my output_log files that the crash generated in case that helps you. Thanks man.

**Comment by HipsterSloth** [comment]
So after digging into that crash some more it appears that it's an issue deep inside libusb trying and failing to acquire the usb device (Specifically getting a failure code when calling WinUSB_Initialize on a device handle for the camera). I couldn't find any documentation online that would explain precisely what would lead to this failure (Error code 87). The two common causes I heard suggested where either the device thinking it was already open by another application or a bad driver associated with the usb device. You might try re-applying the libusb driver to the camera and see if that helps. In the mean time I checked in new 32 and 64 bit binaries (2196bd9482a9fc94ff4bbd35462c403cdce645b4) that handle the camera failure initialization case more gracefully. In your situation it'll still fail to acquire the camera, but at least you won't crash.

**Comment by Ideae** [comment]
Alright thanks for digging man. Must be frustrating relying on such strange
dependencies. I'll just stick to the 64-bit build since that decided to
work for now.

By the way, any ideas of when that new revised package for psmove will be
coming out? I think I heard about that sometime...

**Comment by HipsterSloth** [comment]
@hazem17 I you get a chance would you mind syncing latest and seeing if either the 32 or 64 bit builds work for you.

@Ideae Yeah libusb can bit frustrating to work with sometimes, but it's the best option for cross platform usb support at the moment (that I know about anyway). 

As for the revised package for psmoveapi (a.k.a. the PSMoveService), it's hard to say when it'll be done as myself and cboulay can only work on it our free time, which has been in fits and spurts lately. We currently have a service framework running that connects to psmove controllers, ps3eye cameras, and an HMD and then networks the data to either a C++ client dll plugin or to a config client that combines all of the calibration tools into one visual tool. That said there is still a fair bit of work remaining including: porting over the positional tracking (not doing anything with the video frames read from the PS3EYE at the moment), HMD/PS3EYE coregistration, and updating psmove-unity5 and psmove-ue4 to use the new client dll. Rather than re-write the Unity and UE4 plugins I'm thinking about making an OpenVR plugin instead, which I think would eliminate the need to a separate psmove plugin (since both Unity and Unreal have OpenVR tracked controller support). At minimum it's probably another month of work before we can have something publicly available. That's actually why I'm working on multi-cam positional tracking now in psmoveapi rather than the new service since people can try it out sooner.

**Comment by Ideae** [comment]
An OpenVR plugin sounds great! I'm actually waiting to get my Vive right now. When that happens, I'll be moving over to OpenVR (SteamVR), so I can have both the Vive and DK2 work. When the Vive is available, obviously my application would use the Vive controllers, but if only the DK2 is available, it would be nice to have the PSMove controllers also work!

**Comment by hazem17** [comment]
The 64 bit build works great, but the 32 bit still crashes, i used unity 5.3.1 64 bit to build the project.

**Comment by HipsterSloth** [comment]
@hazem17 Does the 32-bit version of ps3eye_sdl.exe crash for you as well in the assets/plugins/x86 folder? If so is there an error code in the output window it gives? I'm wondering if you get the same libusb error message Ideae was getting.

**Comment by hazem17** [comment]
yes is crashes and gives me this error message: libusb: error [winusbx_claim_interface] could not access interface 0: [87] The parameter is incorrect.
